proof-defns/proof_defns.tex

```diff
@@ -0,0 +1,438 @@
+    For some value of $c$, we say that a transformation $T$ is $c$\textit{-stable} if for all $x, x'$ in the input domain $\X$, and for input metric $d_{\X}$ and output metric $d_{\Y}$,
+    \begin{equation}
+        d_{\mathcal{Y}}(T(x), T(x')) \leq c \cdot d_{\mathcal{X}}(x, x').
+    \end{equation}
+    We say that $c$ is the \textit{stability parameter} of $T$.
+\end{definition}
+
+\silvia{Add a sentence saying that not all stability relations can be expressed only with $c$ (linear relationship).}
+
+In the Rust code, the stability parameter $c$ (which in the case of clamping is equal to~1) gets wrapped up inside of the stability relation property, and the end user can test it empirically. \todo{Explain better -- but perhaps in the general guidelines doc?}
+
+\begin{definition}[Stability relation]
+    The \textit{stability relation}, denoted $\Relation(\din, \dout)$, is a boolean function which takes as input some $\din, \dout$ appropriately quantified and returns \texttt{True} if and only if the relation $\dout \geq c \cdot \din$ for some specified value of $c$. 
+\end{definition}
+
+We can also write the stability relation in the following form:
+
+\begin{equation}
+    \Relation(\din, \dout) = 
+    \begin{cases} 
+      \True & \textrm{if } \dout \geq c \cdot \din \\
+      \False & \textrm{otherwise},
+   \end{cases}
+\end{equation}
+specifying the concrete metrics $d_{\X}, d_{\Y}$ and types of $\din, \dout$ inside the function.
+
+The associated type of $d_{\X}$ is equal to the type of $\din$, and the associated type of $d_{\Y}$ is equal to the type of $\dout$. Importantly, such relations are sound but \textit{not necessarily complete}. A transformation is considered \textit{valid} if its stability relation is sound.
+
+%\begin{definition}[Relation]
+%Relation($d_{in}$, $d_{out}$) means that \grace{Silvia can you add what you had for Relation here?}
+%\end{definition}
+
+\subsection{Stability for Randomness}
+
+
+The following lemma and corollary are used to prove the stability guarantee holds in random transformations like \texttt{make\_impute}. 
+\begin{lemma}[Stability for Randomness]
+We define a randomized function $f: \texttt{DI} \rightarrow \texttt{DO}$. $\Relation(d_{in}, d_{out}) = True$ implies that for all $x, x' \in$ \texttt{DI} that are $d_{in}$-close, there exists a coupling $(R, R')$ of the randomness of $f(x)$ and $f(x')$ such that for all $(r, r') \in Support(R, R'),$ $f_r(x)$ is $d_{out}$-close to $f_{r'}(x')$.
+\end{lemma}
+
+\silvia{Proof? (Grace)}
+
+The following corollary allows us to fix the random seed in random transformations to prove the stability guarantee.
+
+\begin{corollary}
+For randomized function $f: \texttt{DI} \rightarrow \texttt{DO}$, $\Relation(d_{in}, d_{out}) = True$ implies that for all $x, x' \in$ \texttt{DI} that are $d_{in}$-close and for all fixing of the randomness $r$ of $f$ (fixing seed of PRG), we have that $f_r(x)$ and $f_r(x')$ are $d_{out}$-close.
```

Comment:
Kind of a nit, but are all PRNG states accessible from an initial seed? The stability guarantee holds not just for any seed, but for any PRNG state.